packages/lab/src/badge/Badge.tsx

```diff
@@ -60,32 +34,39 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
     window: targetWindow,
   });
 
-  const badgeId = useId();
-  const childId = useId(
-    isValidElement<HTMLAttributes<HTMLElement>>(children)
-      ? children.props?.id
-      : undefined
-  );
+  let valueText = value;
 
-  let badgeContentValue = badgeContent;
-  if (badgeContentValue > max) {
-    badgeContentValue = `${max}+`;
+  if (typeof value === "number" && max) {
+    valueText = value > max ? `${max}+` : value;
+  } else if (typeof value === "string") {
+    valueText = value.length > 4 ? `${value.slice(0, 4)}...` : value;
```

Comment:
I would remove this. Just display the string contents. Truncation can be done with css.